PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -336,6 +337,13 @@ def miniAOD_customizeCommon(process):
     from RecoEgamma.EgammaTools.egammaObjectModifications_tools import makeVIDBitsModifier
     egamma_modifications.append(makeVIDBitsModifier(process,"egmGsfElectronIDs","egmPhotonIDs"))
 
+    #e/gamma scale & smearing
+    #we will run in value map producting mode keyed to the orginal collection
+    from RecoEgamma.EgammaPhotonProducers.reducedEgamma_tools import calibrateReducedEgamma
+    run2_miniAOD_94XFall17.toModify(process,calibrateReducedEgamma)
+    from RecoEgamma.EgammaTools.egammaObjectModifications_tools import appendReducedEgammaEnergyScaleAndSmearingModifier
+    run2_miniAOD_94XFall17.toModify(egamma_modifications,appendReducedEgammaEnergyScaleAndSmearingModifier)
```

Comment:
Is it true that all variables are completely meaningless if computed by default for all configurations?
I'm trying to avoid solutions that are constructed to work only once for some special case.

Reply:
the normal rule for using process modifiers is to define them where the module is originally defined. 
This applies to both of the .toModify calls above, which are violating this rule.
The presence of multiple in-place modifications to modules in the miniAOD_tools.py can make application of this rule impossible. However, at a first glance, it seems like both egamma_modifications and reducedEgamma modifications are unconditional and can be made in the corresponding cff and cfi files that have the original definition of these objects.

Reply:
so the scale & smearing corrections only apply to a very specific dataset. In this case, the variables will be meaningless if they are applied anything other than the re-reco of 2017 data and Fall17 MC.

For example legacy needs a different set of corrections. I agree this should be handled by global tags but this is where we are.

Reply:
yes, they are completely unconditional and can be removed to other files. Done